hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/FSLeafQueue.java

```diff
@@ -557,28 +599,33 @@ private boolean preemptContainerPreCheck() {
         getFairShare());
   }
 
-  /**
-   * Is a queue being starved for its min share.
-   */
-  @VisibleForTesting
-  boolean isStarvedForMinShare() {
-    return isStarved(getMinShare());
+  private Resource minShareStarvation() {
```

Comment:
You are right that:

starvation = desiredShare - currentUsage;
starved = starvation > 0;

However, you are missing desiredShare = min(minShare, demand). Essentially, the missing case is when the demand is less than minShare but still more than current allocation. 

That said, I see a minor simplification. Will do that.